src/mkdocs_autorefs/references.py

```diff
@@ -197,13 +202,78 @@ def fix_refs(html: str, url_mapper: Callable[[str], str]) -> tuple[str, list[str
     return html, unmapped
 
 
+class AnchorScannerTreeProcessor(Treeprocessor):
+    """Tree processor to scan and register HTML anchors."""
+
+    _htags: ClassVar[set[str]] = {"h1", "h2", "h3", "h4", "h5", "h6"}
+
+    def __init__(self, plugin: AutorefsPlugin, md: Markdown | None = None) -> None:
+        """Initialize the tree processor.
+
+        Parameters:
+            plugin: A reference to the autorefs plugin, to use its `register_anchor` method.
+        """
+        super().__init__(md)
+        self.plugin = plugin
+        self._slug = md.treeprocessors["toc"].slugify
+
+    def run(self, root: Element) -> None:  # noqa: D102
+        if self.plugin.current_page is not None:
+            self._scan_anchors(root)
+
+    def _scan_anchors(self, parent: Element) -> list[str]:
+        ids = []
+        # We iterate on pairs of elements, to check if the next element is a heading (alias feature).
+        for el, next_el in zip_longest(parent, parent[1:], fillvalue=Element("/")):
+            if el.tag == "a":
+                # We found an anchor. Record its id if it has one.
+                if anchor_id := el.get("id"):
```

Comment:
**By the way.** Do we want to make any `a` tag be eligible to become an alias **or** only ones without text and without href?

Reply:
I'd say only the ones without text and without href. Otherwise a link at the end of a paragraph before a heading would create an alias I think?

Reply:
OK that'll be commit https://github.com/oprypin/mkdocs-autorefs/commit/a6a785a254369361ea9a9fd9193bde7932f3255b